src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -254,9 +328,67 @@ void EIRPlantLoopHeatPump::doPhysics(EnergyPlusData &state, Real64 currentLoad)
 
     // get setpoint on the load side outlet
     Real64 loadSideOutletSetpointTemp = this->getLoadSideOutletSetPointTemp(state);
+    Real64 originalLoadSideOutletSPTemp = loadSideOutletSetpointTemp;
+
+    // add free cooling at some point, compressor is off during free cooling, temp limits restrict free cooling range
+
+    Real64 capacityModifierFuncTemp = 1.0;
+    Real64 availableCapacity = this->referenceCapacity;
+    Real64 partLoadRatio = 0.0;
+    bool waterTempExceeded = false;
 
     // evaluate capacity modifier curve and determine load side heat transfer
-    Real64 capacityModifierFuncTemp = Curve::CurveValue(state, this->capFuncTempCurveIndex, loadSideOutletSetpointTemp, this->sourceSideInletTemp);
+    // any adjustment to outlet water temp set point requires some form of iteration
+    for (int loop = 0; loop < 2; ++loop) {
+        capacityModifierFuncTemp = Curve::CurveValue(state, this->capFuncTempCurveIndex, loadSideOutletSetpointTemp, this->sourceSideInletTemp);
+
+        availableCapacity = this->referenceCapacity * capacityModifierFuncTemp;
+
+        // apply heating mode dry outdoor (evaporator) coil correction factor for air-cooled equipment
+        if (this->capacityDryAirCurveIndex > 0 && this->airSource && state.dataEnvrn->OutRelHum < RH90) { // above 90% RH yields full capacity
+            Real64 dryCorrectionFactor = std::min(1.0, Curve::CurveValue(state, this->capacityDryAirCurveIndex, state.dataEnvrn->OutDryBulbTemp));
+            if (state.dataEnvrn->OutRelHum <= RH60) {
+                // dry heating capacity correction factor is a function of outdoor dry-bulb temperature
+                availableCapacity *= dryCorrectionFactor;
+            } else {
+                // interpolation of heating capacity between wet and dry is based on outdoor relative humidity over 60%-90% range
+                Real64 semiDryFactor = dryCorrectionFactor + (1.0 - dryCorrectionFactor) * (1.0 - ((RH90 - state.dataEnvrn->OutRelHum) / rangeRH));
```

Comment:
@EnergyArchmage this equation was incorrect. I added the `1.0 -` in `(1.0 - ((RH90 -` to get this math right.